### PR TITLE
Adding support for secrets

### PR DESCRIPTION
**Pull request by nextrevision**

### CODE REVIEW

**Comment by nextrevision** [comment]
Closes #9

File: vars.go [hunk 1]
```diff
@@ -128,21 +129,9 @@ func (vars Vars) toConfigMap(name string, namespace string, convert bool) ([]v1.
...
-				err := fmt.Errorf("%s is not a valid ConfigMap key: %s", v.Key, strings.Join(errs, ", "))
-				return envVars, &v1.ConfigMap{}, err
-			}
-		} else {
-			errs := validation.IsConfigMapKey(v.Key)
-			if len(errs) > 0 {
-				err := fmt.Errorf("%s is not a valid ConfigMap key: %s", v.Key, strings.Join(errs, ", "))
-				return envVars, &v1.ConfigMap{}, err
-			}
+		key, err := validateKey(v.Key, convertKeys)
```
Comment (cllunsford) [inline vars.go]: Use parameter `convert` instead of the global `convertKeys`

File: vars.go [hunk 2]
```diff
@@ -174,3 +163,66 @@ func (vars Vars) toConfigMap(name string, namespace string, convert bool) ([]v1.
 
 	return envVars, configMap, nil
 }
+
+func (vars Vars) toSecret(name string, namespace string, convert bool) ([]v1.EnvVar, *v1.Secret, error) {
+	envVars := []v1.EnvVar{}
+	data := make(map[string][]byte)
+
+	for _, v := range vars {
+		key, err := validateKey(v.Key, convertKeys)
```
Comment (cllunsford) [inline vars.go]: Use the parameter `convert` instead of the global `convertKeys`

**Review by cllunsford — COMMENTED** [review]
Restrict use of global flag vars to the main function.

**Pull request merged**